File: backend/harnesss/cpp-problems/429.cpp
#include <vector>
#include <queue>
using namespace std;

class Solution {
public:
    vector<vector<int>> levelOrder(NaryNode* root) {
        if (!root) return {};
        
        vector<vector<int>> result;
        queue<NaryNode*> q;
        q.push(root);
        
        while (!q.empty()) {
            int currentLevelSize = q.size();
            vector<int> currentLevel;
            
            for (int i = 0; i < currentLevelSize; i++) {
                NaryNode* node = q.front();
                q.pop();
                
                currentLevel.push_back(node->val);
                
                // Push all children into the queue for the next level
                for (NaryNode* child : node->children) {
                    q.push(child);
                }
            }
            result.push_back(currentLevel);
        }
        
        return result;
    }
};

File: backend/harnesss/python-problems/429.py
from typing import List, Optional
from collections import deque

class NaryNode:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children if children is not None else []

class Solution:
    def levelOrder(self, root: Optional['NaryNode']) -> List[List[int]]:
        if not root:
            return []
        
        result = []
        queue = deque([root])
        
        while queue:
            level_size = len(queue)
            current_level = []
            
            for _ in range(level_size):
                node = queue.popleft()
                current_level.append(node.val)
                # NaryNode stores children in a list, so we extend the queue
                queue.extend(node.children)
                
            result.append(current_level)
            
        return result